test/e2e/kubelet_stats.go

```diff
@@ -346,6 +347,7 @@ func computeContainerResourceUsage(name string, oldStats, newStats *cadvisor.Con
 // list of containers, computes and cache resource usage up to
 // maxEntriesPerContainer for each container.
 type resourceCollector struct {
+	lock            sync.RWMutex
```

Comment:
I don't like exposing Lock/Unlock methods to the callers as it may lead to misuse :)